projects/prison-custody-status-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/person/entity/Person.kt

```diff
@@ -19,10 +20,54 @@ class Person(
     @Column(columnDefinition = "char(7)")
     val nomsNumber: String,
 
+    @OneToMany(mappedBy = "person")
+    val additionalIdentifier: List<AdditionalIdentifier> = emptyList(),
+
+    @OneToOne(mappedBy = "person")
+    val manager: PersonManager? = null,
+
     @Column(updatable = false, columnDefinition = "number")
     val softDeleted: Boolean = false
 )
 
 interface PersonRepository : JpaRepository<Person, Long> {
     fun findByNomsNumberAndSoftDeletedIsFalse(nomsNumber: String): List<Person>
+
+    @Query(
+        """
+        SELECT p FROM Person p 
+        LEFT JOIN p.manager
+        JOIN p.additionalIdentifier ai
+        WHERE ai.id = :id 
```

Comment:
Should this be checking the person's id?
```suggestion
        WHERE ai.person.id = :id 
```

Also may be worth returning whether the count is greater than 0, rather than returning the whole Person graph?

And do you need the active manager check?

Reply:
took the query from another project where the same thing is being used, happy to use a count.  if we do not need to check for an active manager i can remove

Reply:
I don't think we need to check for an active manager, we ideally just want a query that returns true/false if the case has been merged.

Reply:
The updated code is still checking the additional_identifier_id instead of the offender_id column, and still joining on manager.  There are also a few unnecessarily mapped columns in the AdditionalIdentifier entity - you can simplify the whole thing quite a bit... this should work:

```kotlin

    @Query(
        """
        select count(ai) > 0 
        from AdditionalIdentifier ai
        where ai.personId = :personId
        and ai.name.code = 'MFCRN'             
        """
    )
    fun personHasBeenMerged(personId: Long): Boolean
}

@Entity
@Immutable
@SQLRestriction("soft_deleted = 0")
class AdditionalIdentifier(
    @Id
    @Column(name = "additional_identifier_id")
    val id: Long,

    @Column(name = "offender_id")
    val personId: Long,

    @ManyToOne
    @JoinColumn(name = "identifier_name_id")
    val name: ReferenceData,

    @Column(updatable = false, columnDefinition = "number")
    val softDeleted: Boolean = false,
)

```